k2/python/tests/ragged_test.py

```diff
@@ -56,23 +56,24 @@ def test_ragged_int_from_str(self):
             self.assertEqual([2, 3], ragged_int.tot_sizes())
 
     def test_pickle_ragged(self):
-        # test num_axes == 2
-        raggeds = ("[ ]", "[ [ ] ]", "[ [1 2] [3] ]")
-        for s in raggeds:
-            ragged_int = k2.RaggedInt(s)
-            torch.save(ragged_int, "ragged.pt")
-            ragged_reload = torch.load("ragged.pt")
-            self.assertEqual(ragged_int, ragged_reload)
-            os.remove("ragged.pt")
+        for device in self.devices:
+            # test num_axes == 2
+            raggeds = ("[ ]", "[ [ ] ]", "[ [1 2] [3] ]")
+            for s in raggeds:
+                ragged_int = k2.RaggedInt(s).to(device)
+                torch.save(ragged_int, "ragged.pt")
+                ragged_reload = torch.load("ragged.pt")
+                self.assertEqual(ragged_int, ragged_reload)
+                os.remove("ragged.pt")
 
-        # test num_axes == 3
-        raggeds = ("[ [ [ ] ] ]", "[ [ [1 2] [3] ] [ [4 5] [6] ] ]")
-        for s in raggeds:
-            ragged_int = k2.RaggedInt(s)
-            torch.save(ragged_int, "ragged.pt")
-            ragged_reload = torch.load("ragged.pt")
-            self.assertEqual(ragged_int, ragged_reload)
-            os.remove("ragged.pt")
+            # test num_axes == 3
+            raggeds = ("[ [ [ ] ] ]", "[ [ [1 2] [3] ] [ [4 5] [6] ] ]")
+            for s in raggeds:
+                ragged_int = k2.RaggedInt(s).to(device)
+                torch.save(ragged_int, "ragged.pt")
+                ragged_reload = torch.load("ragged.pt")
```

Comment:
So I assume map_location would naturally work for this?

Reply:
Yes.
![image](https://user-images.githubusercontent.com/11765074/129700362-339df0d6-8445-4fc2-ab71-1cb7d7275849.png)